SpaceDash/View/CustomView/CustomPopUpView.swift

```diff
@@ -0,0 +1,177 @@
+//  Copyright © 2020 Pushpinder Pal Singh. All rights reserved.
+//
+
+import UIKit
+
+class CustomPopUpView: UIView {
+    
+    var delegate:RocketViewController?{
+        didSet{
+            closeBtn.addTarget(delegate, action: #selector(RocketViewController.closeBtnTappped), for: .touchUpInside)
+        }
+    }
+    
+    ///Demo data
+    let imgArr = ["test_image", "test_image" , "test_image"]
+    
+    ///  Collectionview for Multiple images
+    lazy var collectionView:UICollectionView = {
+        let layout:UICollectionViewFlowLayout = UICollectionViewFlowLayout.init()
+        layout.scrollDirection = .horizontal
+        let cv = UICollectionView(frame: CGRect.zero, collectionViewLayout: UICollectionViewLayout.init())
+        cv.translatesAutoresizingMaskIntoConstraints = false
+        cv.showsHorizontalScrollIndicator = false
+        cv.register(ImageCollectionViewCell.self, forCellWithReuseIdentifier: "ImageCollectionViewCell")
+        cv.setCollectionViewLayout(layout, animated: false)
+        cv.delegate = self
+        cv.dataSource = self
+        cv.isPagingEnabled = true
+        cv.backgroundColor = UIColor.init(named: Constants.Colors.DashCream)
+        cv.bounces = false
+        return cv
+    }()
+    
+    /// Rocket Description
+    public let descriptionText:UILabel = {
+        let l = UILabel()
+        l.translatesAutoresizingMaskIntoConstraints = false
+        l.numberOfLines = 0
+        l.textColor = .black
+        l.text = "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged."
+        return l
+    }()
+    
+    /// Page controlls
+    private let pageControl:UIPageControl = {
+        let c = UIPageControl()
+        c.translatesAutoresizingMaskIntoConstraints = false
+        c.currentPage = 0
+        return c
+    }()
+    
+    /// Close the modal
+    private let closeBtn:UIButton = {
+        let btn = UIButton()
+        btn.setBackgroundImage(UIImage(systemName: "xmark.circle.fill")?.withRenderingMode(.alwaysTemplate), for: .normal)
+        btn.tintColor = UIColor.init(named: Constants.Colors.DashCream)
+        btn.translatesAutoresizingMaskIntoConstraints = false
+        return btn
+    }()
+    
+    /// Divider Line
+    private let dividerLine:UIView = {
+        let v = UIView()
+        v.translatesAutoresizingMaskIntoConstraints = false
+        v.backgroundColor = UIColor(white: 0, alpha: 0.2)
+        return v
+    }()
+    
+    /// Wikepedia btn
+    private let wekipediaBtn:UIButton = {
+        let btn = UIButton()
+        btn.translatesAutoresizingMaskIntoConstraints = false
+        btn.setTitle("Wekipedia", for: .normal)
```

Comment:
typo in Title. Also make a constant under constants file under rocket viewcontroller. Also when clicked, it should open a WKWebView for that rocket, you can just open a demo URL.

Reply:
I don't understand what you mean by constant here, I mean what needs to add in the constants file?

Reply:
Done!